### PR TITLE
Feat/improve new views

### PR DESCRIPTION
**Pull request by lucasbordeau**
- Added scope and family id injection in recoil scoped states
- Added a util to do the same in useRecoilCallbacks
- Added a util to abstract snapshot get value

### CODE REVIEW

File: front/src/modules/views/hooks/useViewInjectedStates.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  const currentView = useRecoilValue(currentViewScopedSelector(scopeId));
+
+  const viewInjectedStates = getViewInjectedStates({
+    viewScopeId: scopeId,
+    familyItemId,
+  });
+
+  return {
+    currentViewId,
```
Comment (charlesBochet) [inline front/src/modules/views/hooks/useViewInjectedStates.ts:43]: remove currentViewdId, and currentView

File: front/src/modules/views/hooks/useViewInjectedStates.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    currentView,
+    viewInjectedStates,
```
Comment (charlesBochet) [inline front/src/modules/views/hooks/useViewInjectedStates.ts:45]: add currentView and currentViewId to returned states of viewInjectedStates

File: front/src/modules/views/utils/getViewSnapshotInjectedStates.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    currentViewScopedSelector(viewScopeId),
+  );
+
+  const viewInjectedStates = getViewInjectedStates({
+    viewScopeId,
+    familyItemId,
+  });
+
+  return {
+    currentViewId,
```
Comment (charlesBochet) [inline front/src/modules/views/utils/getViewSnapshotInjectedStates.ts:44]: remove currentViewId and currentView

File: front/src/modules/views/utils/getViewInjectedStates.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+import { canPersistViewSortsScopedFamilySelector } from '../states/selectors/canPersistViewSortsScopedFamilySelector';
+import { savedViewFieldByKeyScopedFamilySelector } from '../states/selectors/savedViewFieldByKeyScopedFamilySelector';
+import { savedViewFiltersByKeyScopedFamilySelector } from '../states/selectors/savedViewFiltersByKeyScopedFamilySelector';
+import { savedViewSortsByKeyScopedFamilySelector } from '../states/selectors/savedViewSortsByKeyScopedFamilySelector';
+import { viewEditModeScopedState } from '../states/viewEditModeScopedState';
+import { viewObjectIdScopeState } from '../states/viewObjectIdScopeState';
+import { viewsScopedState } from '../states/viewsScopedState';
+import { viewTypeScopedState } from '../states/viewTypeScopedState';
+
+export const getViewInjectedStates = ({
```
[the remark this answers is not in the document]
Comment (charlesBochet) [reply front/src/modules/views/utils/getViewInjectedStates.ts:28]: move this one to utils/internal/

File: front/src/modules/ui/utilities/recoil-scope/utils/getSnapshotValue.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { RecoilState, RecoilValueReadOnly, Snapshot } from 'recoil';
+
+export const getSnapshotStateValue = <StateType>(
```
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply front/src/modules/ui/utilities/recoil-scope/utils/getSnapshotValue.ts:3]: getSnapshotScopedValue ?

File: front/src/modules/views/hooks/useViewInjectedStates.ts (hunk 1 above)
Comment (lucasbordeau) [reply front/src/modules/views/hooks/useViewInjectedStates.ts]: We have to have currentViewId but I removed currentView which is optional

**Review by charlesBochet — APPROVED** [review]
Way better!

**Pull request merged**